apps/central-scan/frontend/src/components/delete_batch_modal.test.tsx

```diff
@@ -0,0 +1,82 @@
+import React from 'react';
+import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
+import userEvent from '@testing-library/user-event';
+import { suppressingConsoleOutput } from '@votingworks/test-utils';
+import { MockApiClient, createMockApiClient } from '../../test/api';
+import { render, screen, waitFor } from '../../test/react_testing_library';
+import { ApiClientContext } from '../api';
+import { DeleteBatchModal } from './delete_batch_modal';
+
+let mockApiClient: MockApiClient;
+
+beforeEach(() => {
+  jest.restoreAllMocks();
+  mockApiClient = createMockApiClient();
+});
+
+afterEach(() => {
+  mockApiClient.assertComplete();
+});
+
+test('allows canceling', async () => {
+  const onClose = jest.fn();
+
+  render(
+    <QueryClientProvider client={new QueryClient()}>
+      <ApiClientContext.Provider value={mockApiClient}>
+        <DeleteBatchModal batchId="a" batchLabel="Batch 1" onClose={onClose} />
+      </ApiClientContext.Provider>
+    </QueryClientProvider>
```

Comment:
Here's an example of how I tried to abstract some of the boilerplate here: https://github.com/votingworks/vxsuite/blob/7a2879e5d28ec6ab43e1750362091c545b58ce78/apps/mark/frontend/test/helpers/mock_api_client.tsx#L150.